spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/intermediate/IRToDnn.scala

```diff
@@ -0,0 +1,214 @@
+  }
+
+  private def fromSpatialMaxPooling(node: IRElement[Float]) : Module[Float] = {
+    val t = node.getOp().asInstanceOf[IRSpatialMaxPooling[Float]]
+    require(t.format == DataFormat.NCHW, "Dnn SpatialMaxPooling only supports NCHW")
+    ReflectUtils.reflectFromIR(node, Class.forName(prefix + "MaxPooling"))
+  }
+
+  private def fromSpatialAveragePooling(node: IRElement[Float]) : Module[Float] = {
+    val t = node.getOp().asInstanceOf[IRSpatialAveragePooling[Float]]
+    require(t.format == DataFormat.NCHW, "Dnn SpatialAveragePooling only supports NCHW")
+    ReflectUtils.reflectFromIR(node, Class.forName(prefix + "AvgPooling"))
+  }
+
+  private def fromSpatialCrossMapLRN(node: IRElement[Float]) : Module[Float] = {
+    val t = node.getOp().asInstanceOf[IRSpatialCrossMapLRN[Float]]
+    require(t.format == DataFormat.NCHW, "Dnn LRN only supports NCHW")
+    ReflectUtils.reflectFromIR(node, Class.forName(prefix + "LRN"))
+  }
+
+  private def fromJoinTable(node: IRElement[Float]) : Module[Float] = {
+    val t = node.getOp().asInstanceOf[IRJoinTable[Float]]
+    require(t.nInputDims == 0,
+      s"Dnn JoinTable only supports nInputDims = 0, but get ${t.nInputDims}")
+    mkldnn.JoinTable(t.dimension)
+  }
+
+  private def fromSpatialBatchNormalization(node: IRElement[Float]) : Module[Float] = {
+    val t = node.getOp().asInstanceOf[IRSpatialBatchNormalization[Float]]
+    require(t.dataFormat == DataFormat.NCHW, "Dnn SpatialBatchNormalization only supports NCHW")
+    val nOutput = t.nOutput
+    val eps = t.eps
+    val momentum = t.momentum
+    val initWeight = t.initWeight
+    val initBias = t.initBias
+    val initGradWeight = t.initGradWeight
+    val initGradBias = t.initGradBias
+
+    val layer = mkldnn.SpatialBatchNormalization(nOutput, eps, momentum,
+      true, initWeight, initBias, initGradWeight, initGradBias)
+
+    val params = node.getParameters()
+    if (params._1 != null) layer.weightAndBias.copy(params._1)
+    if (params._2 != null) layer.gradWeightAndBias.copy(params._2)
+
+    val extraParams = layer.getExtraParameter()
+    if (t.runningMean != null) extraParams(0).copy(t.runningMean.toTensor[Float])
+    if (t.runningVar != null) extraParams(1).copy(t.runningVar.toTensor[Float])
+
+    layer
+  }
+
+  private def fromLinear(node: IRElement[Float]) : Module[Float] = {
+    val t = node.getOp().asInstanceOf[IRLinear[Float]]
+    require(t.wRegularizer == null && t.bRegularizer == null,
+      "Dnn Linear can not support Regularizer")
+    ReflectUtils.reflectFromIR(node, Class.forName(prefix + "Linear"))
+  }
+
+  private def fromBlasModule(node: IRElement[Float]) : Module[Float] = {
+    BlasWrapper(node.getOp().asInstanceOf[IRBlasModuleWrap[Float]].model)
+  }
+
+  private def fromInput(node: IRElement[Float]) : Module[Float] = {
+    mkldnn.Identity[Float]()
+  }
+
+  private def requirement(layer: IRElement[Float]) : Boolean = {
```

Comment:
require?

Reply:
Do you mean change "requirement" to "require"? Should not define "require" func in this scope, otherwise I can not use real require.
 If you think "requirement" is not good, maybe "checkRequirement" ? or you have other good suggestion.

Reply:
checkRequirement is better than requirement

Reply:
ok, changed